msrestazure/azure_active_directory.py

```diff
@@ -525,3 +527,52 @@ def set_token(self, response_url):
             raise_with_traceback(AuthenticationError, "", err)
         else:
             self.token = token
+
+
+# Constants related to AAD-based authentication methods.
+_TOKEN_ENTRY_TOKEN_TYPE = 'tokenType'
+_ACCESS_TOKEN = 'accessToken'
+
+
+class AdalAuthentication(msrest.authentication.Authentication):
+
+    """Base class for adal-derived authentication."""
+
+    def __init__(self, client_id="04b07795-8ddb-461a-bbee-02f9e1bf7b46",
```

Comment:
I'd prefer to have this id in a constant `XPLAT_APP_ID` since it's not recommend anymore to use it:
https://github.com/AzureAD/azure-activedirectory-library-for-python#about-client_id-and-resource-arguments